aikau/src/main/resources/alfresco/documentlibrary/_AlfCreateContentMenuItemMixin.js

```diff
@@ -0,0 +1,182 @@
+       */
+      mimeType: "text/plain",
+
+      /**
+       * This is the topic that will be published when the menu item is clicked. It defaults to the topic requesting
+       * a form dialog be displayed but can be overridden through configuration.
+       *
+       * @instance
+       * @type {String}
+       * @default
+       */
+      publishTopic: "ALF_CREATE_FORM_DIALOG_REQUEST",
+
+      /**
+       * It is expected that upload will require topics to be published globally, however this can be overridden through
+       * configuration.
+       *
+       * @instance
+       * @type {boolean}
+       */
+      publishGlobal: true,
+
+      /**
+       * This should be configured to be an array of form control widgets to be placed into the create content dialog.
+       *
+       * @instance
+       * @type {object[]}
+       * @default null
+       */
+      widgets: null,
+
+      /**
+       * 
+       * @instance
+       */
+      onClick: function alfresco_documentlibrary__AlfCreateContentPermissionsMixin__onClick(/*jshint unused:false*/ evt) {
+         var publishPayload;
+         if (!this.publishPayload)
+         {
+            // If no explicit payload has been configured then use a default payload to generate a form dialog request...
+            publishPayload = {
+               contentWidth: "550px",
+               dialogTitle: this.message(this.dialogTitle),
+               dialogConfirmationButtonTitle: this.message(this.dialogConfirmationLabel),
+               dialogCancellationButtonTitle: this.message(this.dialogCancellationLabel),
+               formSubmissionTopic: this.formSubmissionTopic,
+               formSubmissionPayloadMixin: {
+                  type: this.contentType,
+                  prop_mimetype: this.mimeType || "",
+                  currentNode: this.currentNode,
+                  alfResponseScope: this.pubSubScope
+               },
+               fixedWidth: true,
+               widgets: this.widgets
+            };
+         }
+         else
+         {
+            // ...or use the configured payload
+            publishPayload = this.publishPayload;
+         }
+
+         var publishGlobal = this.publishGlobal || false;
+         var publishToParent = this.publishToParent || false;
+         var payload = this.generatePayload(publishPayload, this.currentItem, null, this.publishPayloadType, this.publishPayloadItemMixin, this.publishPayloadModifiers);
+         this.alfPublish(this.publishTopic, payload, publishGlobal, publishToParent);
+      },
+
+      /**
+       * This handles publications on the 
+       * [metadataChangeTopic]{@link module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#metadataChangeTopic}
+       * topic to set the location of the current folder 
+       * 
+       * @instance
+       * @param {object} payload 
+       */
+      onCurrentNodeChange: function alfresco_documentlibrary__AlfCreateContentPermissionsMixin__onCurrentNodeChange(payload) {
```

Comment:
Incorrect function name for module `alfresco/documentlibrary/_AlfCreateContentMenuItemMixin`